xgi/classes/reportviews.py

```diff
@@ -391,14 +391,4 @@ def members(self, e=None, dtype=list):
         if e not in self:
             raise IDNotFound(f'ID "{e}" not in this view')
 
-        try:
-            return self._id_dict[e].copy()
-        except IDNotFound:
-            if e is None:
-                if dtype is dict:
-                    return {key: self._id_dict[key] for key in self._ids}
-                elif dtype is list:
-                    return [self._id_dict[key] for key in self._ids]
-                else:
-                    raise XGIError(f"Unrecognized dtype {dtype}")
-            raise IDNotFound(f"Item {e} not in this view")
+        return self._id_dict[e].copy()
```

Comment:
I'm p sure it's safe to remove this `copy` call. The problem is that we do not want the returned value to be mutable (or, if it is, we do not want it to mutate the underlying structure). For example, some version the following code can be added to the test suite to make sure that it is safe to do so:
```python
H = xgi.Hypergraph([[1,2,3]])
m = H.edges.members(0)
m.append(10)
assert H.edges.members(0) == [1, 2, 3]
```

Reply:
If it's okay, I was thinking of adding as an issue and dealing with this a bit later. As is, with the changes that I made, we are currently ~2-2.5x slower than a stripped down dict and we added the copy() method because mutating the returned edge will affect the underlying Hypergraph dictionaries.